register.d.ts

```diff
@@ -0,0 +1,10 @@
+import Promise = require('./index');
+
+declare interface Register {
+  Promise: typeof Promise;
+  implementation: string;
+}
+
+declare function register (module?: string): Register;
```

Comment:
`register` now has an options object that can contain either a `Promise` constructor or `global` flag, so would this be something like: 

```typescript
declare interface RegisterOptions {
    Promise?: typeof Promise;
    global?: boolean
}
 declare function register (module?: string, RegisterOptions?): Register; 
```

I completely made that up based on this PR alone.  If I got anywhere close to right, yay for TypeScript!! ;)